src/components/LoginForm.js

```diff
@@ -0,0 +1,137 @@
+import { useState, useEffect } from "react";
+import { useTransition, animated } from "react-spring";
+import { Link, useHistory } from "react-router-dom";
+import { login, validateEmail } from "../service.js";
+import Input from "./Input";
+
+const LoginForm = ({ login }) => {
+    const [user, setUser] = useState({
+        email: "",
+        pass: "",
+    });
+    const [isValid, setIsValid] = useState({
+        email: true,
+        pass: true,
+    });
+
+    const [emailValidError, setEmailValidError] = useState(""); //true if email pattern is invalid
+
+    const [isResponseError, setIsResponseError] = useState(false); //true if login returns an error
+    const [responseError, setResponseError] = useState(false); //contains displayed message of response error
+    const responseErrorAnimation = useTransition(responseError, {
+        from: { opacity: 0 },
+        enter: { opacity: 1 },
+        leave: { opacity: 0 },
+        config: { duration: 300 },
+    });
+
+    let history = useHistory();
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+
+        if (user.email == "" || user.pass == "") {
+            //some validation
+            let email = true,
+                pass = true;
+            if (user.email === "") {
+                email = false;
+                setEmailValidError("To pole nie może być puste!");
+            }
+            if (user.pass === "") {
```

Comment:
Ogólnie co do if, korzystajcie z tego że niektóre wartości są truthy/falsy np. `if (!user.pass)` zamiast `if (user.pass === "")`
Przy większej ilości ifów zwiększa to czytelność.